#!/usr/bin/env python

PACKAGE = "pid_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("kp",    double_t,   0, "gain of proportional unit", 0.001, -1.0, 1.0)
gen.add("ki",    double_t,   0, "gain of integral unit", 0.0, -1.0, 1.0)
gen.add("kd",    double_t,   0, "gain of derivetive unit", 0.0, -1.0, 1.0)
gen.add("intg_len",    int_t,   0, "maximum length of intrgral", 100, 0, 1000)
gen.add("intg_max",    double_t,   0, "maximum volume of intrgral", 0.1, 0.0, 10.0)
gen.add("max_output_v",    double_t,   0, "maximum volume of velocity", 0.5, 0.0, 1.0)
gen.add("max_output_omega",    double_t,   0, "maximum volume of omega", 0.5, 0.0, 1.0)

exit(gen.generate(PACKAGE, "pid_configure", "pid"))
